.github/workflows/build.yml

```diff
@@ -148,135 +165,233 @@ jobs:
           fi
           eval "$shell"
 
-      # TODO signing prototype (see above where "shouldSign" is populated)
+      - name: Generate Checksum
+        id: checksum
+        run: |
+          cd build
+          tar -cvf temp.tar -C temp .
+          echo "sha256=$(sha256sum temp.tar)" >> "$GITHUB_OUTPUT"
+      - name: Stage artifact
+        id: oci
+        uses: actions/upload-artifact@v4
+        with:
+          name: build-oci
+          path: |
+            build/temp.tar*
+          retention-days: 5
```

Comment:
Joel was adamant that to meet the spec we need to have the provenance generation in a separate job which is why we did it that way

Reply:
:wave: It is a bit nuanced and I felt more comfortable generating provenance data in isolation from the build job (to prevent modification). For sure, there is a requirement to isolate access to provenance signing keys from the build job but there is some ambiguity with respect to what process generates the data.

>Accuracy: The provenance MUST be generated by the control plane (i.e. within the trust boundary [identified in the provenance](https://slsa.dev/spec/v1.1/onepage#model)) and not by a tenant of the build platform (i.e. outside the trust boundary), except as noted below.

> The data in the provenance MUST be obtained from the build platform, either because the generator is the build platform or because the provenance generator reads the data directly from the build platform.

source: https://slsa.dev/spec/v1.1/onepage#provenance-generation

Reply:
@mrjoelkamp thank you for these details!!!